common/src/stack/command/stack/commands/sync/firmware/imp_m7520.py

```diff
@@ -0,0 +1,58 @@
+import stack.commands
+from stack.switch.m7800 import SwitchMellanoxM7800
+
+class Implementation(stack.commands.Implementation):
+
+	def run(self, args):
+		switch = args[0]['host']
+		image_name = args[1]
+
+		switch_attrs = self.owner.getHostAttrDict(switch)
+
+		kwargs = {
+			'username': switch_attrs[switch].get('username'),
+			'password': switch_attrs[switch].get('password')
+		}
+
+		kwargs = {k:v for k, v in kwargs.items() if v is not None}
+
+		s = SwitchMellanoxM7800(switch, **kwargs)
+		s.connect()
+		show_images = s.show_images()
+		images_to_delete = show_images['Available images']
+		for image in images_to_delete:
+			s.image_delete(image)
+
+		url = self.get_url_to_drivers(switch, image_name)
+		status = s.image_fetch(url)
+
+		if status != 200:
+			s.disconnect()
+			return
```

Comment:
There should be a command error here, with the HTTP status as well as the url it was unable to fetch from.